explorer-api/src/state.rs

```diff
@@ -1,20 +1,40 @@
-use chrono::{DateTime, Utc};
-use serde::{Deserialize, Serialize};
 use std::fs::File;
 use std::path::Path;
 
+use chrono::{DateTime, Utc};
 use log::info;
+use serde::{Deserialize, Serialize};
 
 use crate::country_statistics::country_nodes_distribution::{
     ConcurrentCountryNodesDistribution, CountryNodesDistribution,
 };
+use crate::mix_node::models::ThreadsafeMixNodeCache;
+use crate::mix_nodes::ThreadsafeMixNodesResult;
+use crate::ping::models::ThreadsafePingCache;
+use mixnet_contract::MixNodeBond;
 
 // TODO: change to an environment variable with a default value
 const STATE_FILE: &str = "explorer-api-state.json";
 
 #[derive(Clone)]
 pub struct ExplorerApiState {
     pub(crate) country_node_distribution: ConcurrentCountryNodesDistribution,
+    pub(crate) mix_nodes: ThreadsafeMixNodesResult,
+    pub(crate) mix_node_cache: ThreadsafeMixNodeCache,
+    pub(crate) ping_cache: ThreadsafePingCache,
+}
+
+impl ExplorerApiState {
+    pub(crate) async fn get_mix_node(&self, pubkey: &str) -> Option<MixNodeBond> {
+        self.mix_nodes
+            .clone()
+            .get()
+            .await
+            .value
+            .iter()
+            .find(|node| node.mix_node.identity_key == pubkey)
```

Comment:
Since we're having to go through the whole vector each time we want to look up each mixnode, maybe vector is not the best data structure for it? Perhaps HashMap would have been better?

Reply:
Yes, it would be - mind if I add that in another PR where I add the geoip resolved location to the bond?

Reply:
Sounds good